src/Utils/matrix_exponential.jl

```diff
@@ -0,0 +1,61 @@
+# currently supported packages
+const SUPPORTED_EXPONENTIAL_PACKAGES = [:ExponentialUtilities, :Expokit]
+
+# =============
+# Global option
+# =============
+
+global exponential_backend = missing  # global state of the exponential backend
+
+function set_exponential_backend!(backend::Module)
+    global exponential_backend = Val(Symbol(backend))
+end
+
+function get_exponential_backend()
+    ismissing(exponential_backend) && error("the exponential backend is not loaded")
+    return exponential_backend
+end
+
+function _expmat(t, A, b)
+    return _expmat(get_exponential_backend(), t, A, b)
+end
+
+# ====================
+# ExponentialUtilities
+# ====================
+
+function load_exponentialutilities()
+return quote
+
+if ismissing(exponential_backend)
+    set_exponential_backend!(ExponentialUtilities)
+end
+
+end end  # quote / load_exponentialutilities
+
+
+function _expmat(::Val{:ExponentialUtilities}, t, A, b)
```

Comment:
nice :+1:  this makes it very easy to add other exponentiation backends than the ones listed in this file; i mean, from the outside, as well as tweaking options (say, now i want to pass `expv` some extra options i can do it by overwriding `_expmat`)

i wonder if we need `SUPPORTED_EXPONENTIAL_PACKAGES` at all; i would rather remove (and also `require`) because if it was not loaded, a nice error message is shown anyways ([here](https://github.com/JuliaReach/LazySets.jl/pull/2907/files#diff-94a0f99d9daf3a4c7fe1a253635aebc5626457a3b734a92b6086c277375d3c2cR15))

Reply:
We do not need it. But the generic error message does not give concrete instructions what to do, so you have to know which packages work. I think it is helpful for new users to see the list of packages that are automatically supported.